### PR TITLE
Functional hpr2

### PR DESCRIPTION
**Pull request by daineAMD**
See [documentation](http://www.netlib.org/lapack/explore-html/d6/dee/group__complex__blas__level2_ga537b8bcd05cbe626ebce0fbc8d66a2d6.html#ga537b8bcd05cbe626ebce0fbc8d66a2d6).

Fairly similar to previous PR for hpr.

### CODE REVIEW

File: library/src/blas2/rocblas_hpr2.hpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
+/* ************************************************************************
+ * Copyright 2016-2020 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#pragma once
+#include "handle.h"
+#include "rocblas.h"
+
+template <typename T, typename U>
```
Comment (YvanMokwinski) [inline library/src/blas2/rocblas_hpr2.hpp]: Is U needed? Is 'T alpha' a problem here?

File: clients/include/testing_hpr2_strided_batched.hpp [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+                                            stride_y,
+                                            dA_1,
+                                            stride_A,
+                                            batch_count);
+        }
+
+        gpu_time_used     = (get_time_us() - gpu_time_used) / number_hot_calls;
+        rocblas_gflops    = batch_count * hpr2_gflop_count<T>(N) / gpu_time_used * 1e6;
+        rocblas_bandwidth = batch_count * (2.0 * N * (N + 1)) / 2 * sizeof(T) / gpu_time_used / 1e3;
+
```
Comment (TorreZuk) [inline clients/include/testing_hpr2_strided_batched.hpp]: again if you wait put in bytes.hpp

File: library/src/blas2/rocblas_hpr2.hpp [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+                                    ptrdiff_t      shift_y,
+                                    rocblas_int    incy,
+                                    rocblas_stride stride_y,
+                                    TPtr           APa,
+                                    ptrdiff_t      shift_A,
+                                    rocblas_stride stride_A)
+{
+    rocblas_int num_threads = hipBlockDim_x * hipBlockDim_y * hipBlockDim_z;
+    if(DIM_X * DIM_Y != num_threads)
+        return; // need to launch exactly the number of threads as template parameters indicate.
```
Comment (TorreZuk) [inline library/src/blas2/rocblas_hpr2.hpp]: Is this a check on the compiler?   Or guard against reuse?

File: library/src/blas2/rocblas_hpr2.hpp (hunk 1 above)
Comment (daineAMD) [reply library/src/blas2/rocblas_hpr2.hpp]: Nope, alpha should be able to be a T, this must have been left from hpr. Will fix.

File: library/src/blas2/rocblas_hpr2_strided_batched.cpp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+                            stridey,
+                            "stride_a",
+                            strideA,
+                            "batch_count",
+                            batch_count);
+        }
+
+        if(uplo != rocblas_fill_lower && uplo != rocblas_fill_upper)
+            return rocblas_status_not_implemented;
+        if(n < 0 || !incx || !incy || batch_count < 0)
```
Comment (TorreZuk) [inline library/src/blas2/rocblas_hpr2_strided_batched.cpp]: same mismatch

File: library/src/blas2/rocblas_hpr2.hpp (hunk 3 above)
Comment (YvanMokwinski) [reply library/src/blas2/rocblas_hpr2.hpp]: I don't like it. No error message, no status, nothing.

Comment (YvanMokwinski) [reply library/src/blas2/rocblas_hpr2.hpp]: Actually, following a discussion with Scott, we should make the kernel crashing.

Comment (daineAMD) [reply library/src/blas2/rocblas_hpr2.hpp]: The check seems redundant to me, could reach out and see why it was initially added into gemv in the first place.

**Pull request merged**